code/script.js

```diff
@@ -1,325 +1,437 @@
+    currentQuestion = {
+      attribute: value,
+      value: true,
+      category: category,
+    };
+  }
 
   currentQuestion = {
     category: category,
-    // value: value
-  }
-}
+    value: value,
+  };
+  console.log(currentQuestion);
+};
+
+//------------------------------------------------------------------------
 
 // This function should be invoked when you click on 'Find Out' button.
-const checkQuestion = () => {
-  const { category, value } = currentQuestion
 
-  // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
-  if (category === 'hair' || category === 'eyes') {
+const checkQuestion = () => {
+  const { category, value } = currentQuestion;
+  let keep = false;
+  // compares question to secretperson.
+  if (category === "hair" || category === "eyes") {
+    keep = value === secret[category];
+  } else if (category === "accessories" || category === "other") {
+    keep = secret[category].includes(value);
+  }
 
-  } else if (category === 'accessories' || category === 'other') {
+  // After determining keep, call filterCharacters with keep as an argument
+  filterCharacters(keep);
+};
 
-  }
-}
+//----------------------------------------------------------------------
 
-// It'll filter the characters array and redraw the game board.
 const filterCharacters = (keep) => {
-  const { category, value } = currentQuestion
+  const { category, value } = currentQuestion;
   // Show the correct alert message for different categories
-  if (category === 'accessories') {
+  if (category === "accessories") {
     if (keep) {
+      charactersInPlay = charactersInPlay.filter((person) =>
+        person[category].includes(value)
+      );
       alert(
-        `Yes, the person wears ${value}! Keep all people that wears ${value}`
-      )
+        `Yes, the person wears ${value}! Keep all people that wear ${value}`
+      );
     } else {
+      charactersInPlay = charactersInPlay.filter(
+        (person) => !person[category].includes(value)
+      );
+
       alert(
-        `No, the person doesn't wear ${value}! Remove all people that wears ${value}`
-      )
+        `No, the person doesn't wear ${value}! Remove all people that wear ${value}`
+      );
     }
-  } else if (category === 'other') {
-    // Similar to the one above
-  } else {
+  } else if (category === "other") {
+    if (keep) {
+      charactersInPlay = charactersInPlay.filter((person) =>
+        person[category].includes(value)
+      );
+
+      alert(
+        `Yes, the person has a ${value}! Keep all people that have a ${value}`
+      );
+    } else {
+      charactersInPlay = charactersInPlay.filter(
+        (person) => !person[category].includes(value)
+      );
+
+      alert(
+        `No, the person doesn't have a ${value}! Remove all people that have a ${value}`
```

Comment:
The alert message says ”Yes, the person has a smoker! Keep all people that have a smoker” Is there a way to change this?